extension/observer/dockerobserver/extension.go

```diff
@@ -36,6 +43,140 @@ func (d *dockerObserver) Shutdown(ctx context.Context) error {
 	return nil
 }
 
+// endpointsForContainer generates a list of observer.Endpoint given a Docker ContainerJSON.
+// This function will only generate endpoints if a container is in the Running state and not Paused.
+func (d *dockerObserver) endpointsForContainer(c *dtypes.ContainerJSON) []observer.Endpoint {
+	cEndpoints := make([]observer.Endpoint, 0)
+
+	if c.State.Running && !c.State.Paused {
+		knownPorts := map[nat.Port]bool{}
+		for k := range c.Config.ExposedPorts {
+			knownPorts[k] = true
+		}
+
+		// iterate over exposed ports and try to create endpoints
+		for portObj := range knownPorts {
+			endpoint := d.endpointForPort(portObj, c)
+			// the endpoint was not set, so we'll drop it
+			if endpoint == (observer.Endpoint{}) {
```

Comment:
Not certain how desirable this zero value check is since some Endpoint fields may not be comparable now or in the future (map of interfaces and an interface field).  May be better to return a pointer and have a nil check?